stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOMStorage.java

```diff
@@ -2,20 +2,185 @@
+
+  public DOMStorage(Context context) {
+    mContext = context;
+    mDOMStoragePeerManager = new DOMStoragePeerManager(context);
   }
 
   @ChromeDevtoolsMethod
   public void enable(JsonRpcPeer peer, JSONObject params) {
+    mDOMStoragePeerManager.addPeer(peer);
   }
 
+  @ChromeDevtoolsMethod
   public void disable(JsonRpcPeer peer, JSONObject params) {
+    mDOMStoragePeerManager.removePeer(peer);
+  }
+
+  @ChromeDevtoolsMethod
+  public JsonRpcResult getDOMStorageItems(JsonRpcPeer peer, JSONObject params)
+      throws JSONException {
+    StorageId storage = mObjectMapper.convertValue(
+        params.getJSONObject("storageId"),
+        StorageId.class);
+
+    ArrayList<List<String>> entries = new ArrayList<List<String>>();
+    String prefTag = storage.securityOrigin;
+    if (storage.isLocalStorage) {
+      SharedPreferences prefs = mContext.getSharedPreferences(prefTag, Context.MODE_PRIVATE);
+      for (Map.Entry<String, ?> prefsEntry : prefs.getAll().entrySet()) {
+        ArrayList<String> entry = new ArrayList<String>(2);
+        entry.add(prefsEntry.getKey());
+        entry.add(SharedPreferencesHelper.valueToString(prefsEntry.getValue()));
+        entries.add(entry);
+      }
+    }
+
+    GetDOMStorageItemsResult result = new GetDOMStorageItemsResult();
+    result.entries = entries;
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public void setDOMStorageItem(JsonRpcPeer peer, JSONObject params) throws JSONException {
+    StorageId storage = mObjectMapper.convertValue(
+        params.getJSONObject("storageId"),
+        StorageId.class);
+    String key = params.getString("key");
+    String value = params.getString("value");
+
+    if (storage.isLocalStorage) {
+      SharedPreferences prefs = mContext.getSharedPreferences(
+          storage.securityOrigin,
+          Context.MODE_PRIVATE);
+      Object exitingValue = prefs.getAll().get(key);
+      SharedPreferences.Editor editor = prefs.edit();
+      if (!tryAssignByType(editor, key, value, exitingValue)) {
+        editor.putString(key, value);
+      }
+      editor.apply();
+    }
+  }
+
+  @ChromeDevtoolsMethod
+  public void removeDOMStorageItem(JsonRpcPeer peer, JSONObject params) throws JSONException {
+    StorageId storage = mObjectMapper.convertValue(
+        params.getJSONObject("storageId"),
+        StorageId.class);
+    String key = params.getString("key");
+
+    if (storage.isLocalStorage) {
+      SharedPreferences prefs = mContext.getSharedPreferences(
+          storage.securityOrigin,
+          Context.MODE_PRIVATE);
+      prefs.edit().remove(key).apply();
+    }
+  }
+
+  private static boolean tryAssignByType(
+      SharedPreferences.Editor editor,
+      String key,
+      String value,
+      @Nullable Object existingValue) {
+    try {
+      if (existingValue instanceof Integer || existingValue instanceof Long) {
+        editor.putLong(key, Long.parseLong(value));
```

Comment:
Oh, good call.  I assumed `Long` was a strict superset of `Integer` for `SharedPreferences` but this is not the case.  This could cause crashes in caller code.

I'll correct it.